sw/device/silicon_creator/rom/e2e/rom_e2e_shutdown_exception_c_test.c

```diff
@@ -0,0 +1,21 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/manifest_def.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+// Note: Since this test program is written to verify the behavior of the ROM C
+// exception handlers,
+// - it does not have a `.crt` section,
```

Comment:
What's the mechanism that decides that this program has no `.crt`section? Is it that `:rom_e2e_shutdown_exception_c` depends on `//sw/device/silicon_creator/rom_ext:ld_slot_a`, which pulls in `bare_metal_common.ld`? That can't be it because I see part of that linker script defines the `.crt `section. https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_owner/bare_metal/bare_metal_common.ld;l=73;drc=da90a2d352825f2e4068941ad2172a40928d8496

Reply:
> What's the mechanism that decides that this program has no `.crt`section?

Because we don't define `.crt` section for this program like the ones defined in `rom_ext_start.S` or `rom_start.S`. From the `ld` manual:

> The linker will not create output sections which do not have any contents. This is for convenience when referring to input sections that may or may not exist. For example,
> ```
> .foo { *(.foo) }
> ```
> will only create a `.foo' section in the output file if there is a `.foo' section in at least one input file.

Reply:
Aha, that makes sense. Thanks!